packages/nuxt/src/pages/utils.ts

```diff
@@ -140,19 +140,19 @@ export function generateRoutesFromFiles (files: ScannedFile[], options: Generate
   return prepareRoutes(routes)
 }
 
-export async function augmentPages (routes: NuxtPage[], vfs: Record<string, string>, augmentedPages = new Set<string>()) {
+export async function augmentPages (routes: NuxtPage[], vfs: Record<string, string>, augmentedPagesWithPaths = new Set<string>(), parentPath = '') {
   for (const route of routes) {
-    if (route.file && !augmentedPages.has(route.file)) {
+    if (route.file && !augmentedPagesWithPaths.has(route.file)) {
```

Comment:
We're not checking if `augmentedPagesWithPaths` has `${route.file}_${parentPath}${route.path}` here, does this mean the route file will always be read?

Reply:
Oh, my bad, yes. You are right. we should check that